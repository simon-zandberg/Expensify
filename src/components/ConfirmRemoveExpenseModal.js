import React from 'react';
import Modal from 'react-modal';

const ConfirmRemoveExpenseModal = (props) => (
    <Modal
        appElement = {document.getElementById('app')}
        isOpen = {!!props.selectedExpense}
        //onRequestClose = {props.handleClearSelectedOption}
        contentLabel = 'Confirm expense removal'
        closeTimeoutMS={200}
        className="modal"
    >
        <h3 className="modal__title">Confirm removal?</h3>
        {props.selectedExpense && <p className="modal__body">{props.selectedExpense.description}</p>}
        <div className="content-container">
            <div className="input-group">
                <div className="input-group__item">        
                    <button className="button" onClick={props.onRemoveExpense}>OK</button>
                </div>
                <div className="input-group__item">
                    <button className="button button--secondary" onClick={props.cancelRemoveExpense}>Cancel</button>
                </div>
            </div>
        </div>
    </Modal>
);

export default ConfirmRemoveExpenseModal;