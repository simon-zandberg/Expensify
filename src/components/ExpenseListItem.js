import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';

const ExpenseListItem = ({ id, description, amount, createdAt }) => {
    return (<tr>
        <td align="left"><Link to={`/edit/${id}`}>{description}</Link></td>
        <td align="right">${(amount / 100).toFixed(2)}</td>
        <td align="right">{moment(createdAt).format("D MMM YYYY")}</td>
    </tr>);
};

export default ExpenseListItem;