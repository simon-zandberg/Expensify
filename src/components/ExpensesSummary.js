import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import expensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpensesSummary = (props) => {
    if (props.expenses.length > 0) {
        return (
            <div>
                <h3>
                    Viewing {props.expenses.length} expenses totalling {numeral(expensesTotal(props.expenses) / 100).format('$0,0.00')}
                </h3>
            </div>
        );
    } else
        return null;
};

const mapStateToProps = (state, props) => ({
    expenses: selectExpenses(state.expenses, state.filters)
});

export default connect(mapStateToProps)(ExpensesSummary);