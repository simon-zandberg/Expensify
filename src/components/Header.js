import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = () => (
    <header>
        <h1>Expensify</h1>
        <div><NavLink to="/" activeClassName="is-active" exact={true}>Dashboard</NavLink></div>
        <div><NavLink to="/create" activeClassName="is-active">Create Expense</NavLink></div>
    </header>
);

export default Header;