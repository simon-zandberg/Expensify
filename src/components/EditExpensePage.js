import React from 'react';
import { connect } from 'react-redux';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';
import ExpenseForm from './ExpenseForm';
import ConfirmRemoveExpenseModal from './ConfirmRemoveExpenseModal';

export class EditExpensePage extends React.Component {
    state = {
        selectedExpense: null
    };

    onSubmit = (expense) => {
        this.props.startEditExpense(this.props.expense.id, expense);
        this.props.history.push('/');
    };

    onRemoveExpense = () => {
        this.props.startRemoveExpense({ id: this.props.expense.id });
        this.props.history.push('/');
    };

    setSelectedExpense = () => {
        this.setState(() => ({ selectedExpense: this.props.expense }));        
    }

    cancelRemoveExpense = () => {
        this.setState(() => ({ selectedExpense: null }));
    }

    render() {
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Edit Expense</h1>
                    </div>
                </div>
                <div className="content-container">
                    <ExpenseForm expense={this.props.expense} onSubmit={this.onSubmit} />
                    <button className="button button--secondary" onClick={this.setSelectedExpense}>Remove Expense</button>
                </div>
                <ConfirmRemoveExpenseModal
                    selectedExpense={this.state.selectedExpense}
                    onRemoveExpense={this.onRemoveExpense}
                    cancelRemoveExpense={this.cancelRemoveExpense}
                />
            </div>
        );
    }
};

const mapStateToProps = (state, props) => ({
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
});

const mapDispatchToProps = (dispatch) => ({
    startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
    startRemoveExpense: (data) => dispatch(startRemoveExpense(data))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);