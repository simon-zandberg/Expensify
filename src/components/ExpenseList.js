import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';

export const ExpenseList = (props) => (
    <div>
        { props.expenses.length === 0 ? (
            <p>No expenses.</p>
        ) : (
            <div>
                <h1>Expense List</h1>
                <table width="100%">
                    <thead>
                        <tr>
                            <th align="left">Description</th>
                            <th align="right">Amount</th>
                            <th align="right">Created At</th>
                        </tr>
                    </thead>
                    <tbody>
                        { props.expenses.map((expense) => {
                            return <ExpenseListItem key={expense.id} {...expense} />;
                        })}
                    </tbody>
                </table>
                <p style={{display:'none'}}>{props.expenses.length}</p>
            </div>
        )}
    </div>
);

const mapStateToProps = state => ({
    expenses: selectExpenses(state.expenses, state.filters)
});

export default connect(mapStateToProps)(ExpenseList);