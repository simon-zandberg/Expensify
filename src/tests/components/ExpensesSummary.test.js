import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should display summary for 2 expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={2} expensesTotal={4564} />);
    expect(wrapper).toMatchSnapshot();
});

test('should display summary for 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={450} />);
    expect(wrapper).toMatchSnapshot();
});

test('should display summary for no expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={0} expensesTotal={0} />);
    expect(wrapper).toMatchSnapshot();
});