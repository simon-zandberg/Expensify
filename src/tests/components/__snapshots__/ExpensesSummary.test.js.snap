// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display summary for 1 expense 1`] = `
<div>
  <h3>
    Viewing 
    1
     
    expense
     totalling 
    $4.50
  </h3>
</div>
`;

exports[`should display summary for 2 expenses 1`] = `
<div>
  <h3>
    Viewing 
    2
     
    expenses
     totalling 
    $45.64
  </h3>
</div>
`;

exports[`should display summary for no expenses 1`] = `
<div>
  <h3>
    Viewing 
    0
     
    expenses
     totalling 
    $0.00
  </h3>
</div>
`;
